uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-assignment.controller.js

```diff
@@ -270,32 +357,32 @@ define([
 
         var activity;
```

Comment:
Great that you refactored this function. But still this looks horrible, can we go a little bit further to refactor it please
1. Merge `documentList` and `documentType` into
`document = { type: '', list: []}`
Same for task.

2.Break `documentType = activity.name ? $filter('filter')($rootScope.cache.documentType.arr, { value: activity.name }, true)[0] : '';` line into multiple lines. Its too long now. Same for `taskType`.

3. Move the content for `activity.activitySet` watcher, in a separate function, as it has so much content.